nemo/collections/nlp/data/language_modeling/text_memmap_dataset.py

```diff
@@ -207,6 +208,28 @@ def _build_data_from_text(self, text):
         return super()._build_data_from_text(text)
 
 
+class JSONLMemMapDataset(TextMemMapDataset):
+    """
+    Memory-mapped iteration over a JSONL file.
+    """
+
+    def __init__(
+        self, dataset_paths, newline_int=10, header_lines=1, workers=None, tokenizer=None, sort_dataset_paths=True,
+    ):
+        super().__init__(
+            dataset_paths=dataset_paths,
+            newline_int=newline_int,
+            header_lines=header_lines,
+            workers=workers,
+            tokenizer=tokenizer,
+            sort_dataset_paths=sort_dataset_paths,
+        )
+
+    def _build_data_from_text(self, text):
```

Comment:
The assumption here is that the json file is formatted with 1 data point per line, right?
Also, good to see it is easy to extend :)

Reply:
Yes!